src/dns.ml

```diff
@@ -1303,6 +1303,180 @@ module Nsec3 = struct
+    let arcsecs, dir = arcsecs_print lat in
+    let dir = if dir then "N" else "S" in
+    (arcsecs, dir)
+
+  let long_print long =
+    let arcsecs, dir = arcsecs_print long in
+    let dir = if dir then "E" else "W" in
+    (arcsecs, dir)
+
+  let alt_print alt =
+    let (+), (-), (/) = Int64.add, Int64.sub, Int64.div in
+    (* convert a uint32 alt to an int64 *)
+    let alt = if alt < 0l then
+        Int64.of_int32 alt + Int64.shift_left 1L 32
+      else Int64.of_int32 alt
+    in
+    let alt = alt - 10000000L in
+    (alt / 100L, Int64.rem alt 100L)
+  
+  let precision_print prec =
+    let mantissa = ((Int.shift_right prec 4) land 0x0f) mod 10 in
+    let exponent = ((Int.shift_right prec 0) land 0x0f) mod 10 in
+    let (/), ( * ) = Int64.div, Int64.mul in
+    let p = Int64.of_int mantissa * pow10 exponent in
+    (p / 100L, Int64.rem p 100L)
+
+  let to_string loc =
+    let decimal_string (integer, decimal) decimal_digits =
+      let (/), ( * ) = Int64.div, Int64.mul in
+      let integer_string = Int64.to_string integer in
+      if decimal = 0L then
+        integer_string
+      else
+      let decimal_string =
+        let rec trim_trailing_zeros decimal num_trimmed =
+          if (decimal / 10L) * 10L = decimal then
+            trim_trailing_zeros (decimal / 10L) (num_trimmed + 1)
+          else
+            decimal, num_trimmed
+        in
+        (* remove trailing zero from decimal *)
+        let decimal, num_trimmed = trim_trailing_zeros decimal 0 in
+        (* left pad zeros *)
+        let decimal = Int64.to_string decimal in
+        String.make (decimal_digits - String.length decimal - num_trimmed) '0' ^ decimal
+      in
+      integer_string ^ "." ^ decimal_string
+    in
+    let lat_long_to_string deg min sec dir =
+      let sec_string =
+        let integer, decimal = sec in
+        decimal_string (Int64.of_int32 integer, Int64.of_int32 decimal) 3
+      in
+      String.concat " " ((List.map (Int32.to_string) [deg; min]) @ [sec_string ; dir]) 
+    in
+    let lat_string =
+      let (lat_deg, lat_min, lat_sec), lat_dir = lat_print loc.latitude in
+      lat_long_to_string lat_deg lat_min lat_sec lat_dir
+    in
+    let long_string =
+      let (long_deg, long_min, long_sec), long_dir = long_print loc.longitude in
+      lat_long_to_string long_deg long_min long_sec long_dir
+    in
+    let meter_values =
+      List.map (fun m -> decimal_string m 2 ^ "m") (
+        [alt_print loc.altitude] @ (List.map precision_print [loc.size; loc.horiz_pre; loc.vert_pre])    
+      )
+    in
+    String.concat " " ([lat_string; long_string;] @ meter_values)
+
+  let pp ppf loc = Fmt.pf ppf "LOC %s" (to_string loc)
+
+  let compare a b =
+    List.fold_right andThen [
+      compare a.latitude b.latitude ;
+      compare a.longitude b.longitude ;
+      compare a.altitude b.altitude ;
+      compare a.size b.size ;
+      compare a.horiz_pre b.horiz_pre ;
+      compare a.vert_pre b.vert_pre ;
+    ] 0
```

Comment:
these use the polymorphic compare function all over... and could just be written as `compare a b` -- but the polymorphic compare isn't nice... so can we use `int32_compare` (defined in `dns.ml`) and `int_compare` respectively?

Reply:
oh thanks, I opened #314 to avoid polymorphic compare usage.